### PR TITLE
qa/rgw: add multisite suite to configure and run multisite tests

### PR DESCRIPTION
**Pull request by cbodley**
work in progress for a multisite test suite in teuthology

TODO:
- [x] create zone pools depending on `rgw.ec_data_pool`
- [x] apply `rgw compression type` setting for created zones
- [x] add `rgw_multisite_tests` task to run multisite tests using configuration from `rgw-multisite` task
- [x] add `nose` as dependency to teuthology (see https://github.com/ceph/teuthology/pull/1069)

### CODE REVIEW

**Comment by cbodley** [comment]
three-zone tests are currently [failing](http://pulpito.ceph.com/cbodley-2017-05-01_10:10:54-rgw-master---basic-smithi/), testing https://github.com/ceph/ceph/pull/14915 as a potential fix

File: qa/tasks/rgw.py [hunk 1]
```diff
@@ -221,6 +218,8 @@ def ship_apache_configs(ctx, config, role_endpoints, on_client = None,
     finally:
         log.info('Removing apache config...')
         for client in clients_to_create_as:
+            cluster_name, daemon_type, client_id = teuthology.split_role(client)
+            client_with_cluster = '.'.join((cluster_name, daemon_type, client_id))
```
Comment (oritwas) [inline qa/tasks/rgw.py]: can you add some logging?

File: qa/tasks/rgw.py [hunk 2]
```diff
@@ -1121,53 +1114,26 @@ def task(ctx, config):
     overrides = ctx.config.get('overrides', {})
     teuthology.deep_merge(config, overrides.get('rgw', {}))
 
-    regions = {}
-    if 'regions' in config:
-        # separate region info so only clients are keys in config
-        regions = config['regions']
-        del config['regions']
+    regions = config.pop('regions', {})
```
Comment (oritwas) [inline qa/tasks/rgw.py]: maybe it is time to change it to zonegroups :)

File: src/test/rgw/rgw_multi/tests.py [hunk 3]
```diff
@@ -97,6 +89,17 @@ def meta_sync_status(zone):
 
     return period, realm_epoch, num_shards, markers
 
+def meta_sync_status(zone):
+    for _ in range(60):
```
Comment (oritwas) [inline src/test/rgw/rgw_multi/tests.py]: can you add a comment why 60?

File: src/test/rgw/rgw_multi/tests.py [hunk 4]
```diff
@@ -300,49 +308,54 @@ def zone_data_checkpoint(target_zone, source_zone):
...
+                  (target_zone.name, source_zone.name, bucket_name)
 
 def set_master_zone(zone):
     zone.modify(zone.cluster, ['--master'])
     zonegroup = zone.zonegroup
     zonegroup.period.update(zone, commit=True)
     zonegroup.master_zone = zone
     # wait for reconfiguration, so that later metadata requests go to the new master
-    time.sleep(5)
+    time.sleep(15)
```
Comment (oritwas) [inline src/test/rgw/rgw_multi/tests.py]: maybe it should be configurable?

File: qa/tasks/rgw.py (hunk 2 above)
Comment (cbodley) [inline qa/tasks/rgw.py]: yeah, i have a https://github.com/cbodley/ceph/commits/wip-rgw-agent-cleanup branch that removes this stuff

File: src/test/rgw/rgw_multi/tests.py (hunk 4 above)
Comment (cbodley) [inline src/test/rgw/rgw_multi/tests.py]: good idea! then we can keep it at 5 for test_multi.py, and raise it as much as we need to for teuthology

File: src/test/rgw/rgw_multi/tests.py (hunk 3 above)
Comment (cbodley) [inline src/test/rgw/rgw_multi/tests.py]: yeah, i just settled on a total of 5 minutes. i'll make this configurable along with the sleep in `set_master_zone()`

File: qa/tasks/rgw.py (hunk 1 above)
Comment (cbodley) [inline qa/tasks/rgw.py]: what would you like to see logged here? just that it's getting the right value for `client_with_cluster`? here's a snippet from a [teuthology.log](http://qa-proxy.ceph.com/teuthology/cbodley-2017-05-16_10:58:37-rgw-master---basic-mira/1185156/teuthology.log) that shows this in action:
> INFO:tasks.rgw.client.0:Stopped
INFO:teuthology.orchestra.run.mira114:Running: 'rm -f /home/ubuntu/cephtest/rgw.opslog.ceph.client.0.sock'
INFO:tasks.rgw:Stopping apache...
INFO:tasks.rgw:Removing apache config...
INFO:teuthology.orchestra.run.mira114:Running: 'rm -f /home/ubuntu/cephtest/apache/apache.ceph.client.0.conf && rm -f /home/ubuntu/cephtest/apache/htdocs.ceph.client.0/rgw.fcgi'
INFO:tasks.rgw:Cleaning up apache directories...
INFO:teuthology.orchestra.run.mira114:Running: 'rm -rf /home/ubuntu/cephtest/apache/tmp.ceph.client.0 && rmdir /home/ubuntu/cephtest/apache/htdocs.ceph.client.0'
INFO:teuthology.orchestra.run.mira114:Running: 'rmdir /home/ubuntu/cephtest/apache'
DEBUG:teuthology.run_tasks:Unwinding manager ceph

Comment (oritwas) [inline qa/tasks/rgw.py]: I wanted the cluster name and client id.
I see from the logs the client id is already logged and I can find out what is cluster name from the teuthology node. 
No need to add more logging

**Comment by cbodley** [comment]
the latest run of the full rgw suite with `--subset=1/6 --filter frontend/civetweb.yaml` was all green: http://pulpito.ceph.com/cbodley-2017-05-17_14:49:58-rgw-master---basic-smithi/

the rgw:multisite suite pretty much always fails with valgrind errors (and [osd crashes on shutdown](http://tracker.ceph.com/issues/19931)), but the multisite tests themselves are passing:
http://pulpito.ceph.com/cbodley-2017-05-16_10:46:23-rgw:multisite-master---basic-smithi/

> INFO:tasks.rgw_multisite_tests:----------------------------------------------------------------------
INFO:tasks.rgw_multisite_tests:Ran 11 tests in 921.748s
INFO:tasks.rgw_multisite_tests:
INFO:tasks.rgw_multisite_tests:OK

**Pull request merged**